core/src/apps/bitcoin/common.py

```diff
@@ -179,3 +180,20 @@ def tagged_hashwriter(tag: bytes) -> HashWriter:
     ctx = sha256(tag_digest)
     ctx.update(tag_digest)
     return HashWriter(ctx)
+
+
+def format_fee_rate(
+    fee_rate: float, coin: CoinInfo, include_shortcut: bool = False
+) -> str:
+    # Use format_amount to get correct thousands separator -- micropython's built-in
+    # formatting doesn't add thousands sep to floating point numbers.
+    # We multiply by 100 to get a fixed-point integer with two decimal places,
+    # and add 0.5 to round to the nearest integer.
+    fee_rate_formatted = format_amount(int(fee_rate * 100 + 0.5), 2)
```

Comment:
Nit: why not `math.ceil`?

Reply:
First of all, this is a refactor - the original code already exists in the master branch.

Second of all, `math.ceil` is not what `int(x + 0.5)` does. I guess you meant `round`? 

The original code was added in a41cfadbf1eff82e1e25c1ab7b960e73f1f0cdfa by @marnova. Martin: Did you have any special reason why not to use just `round(fee_rate, 2)`? We don't use this anywhere in the codebase, so there might be a reason why not to use it, but I don't know of any.

Reply:
First of all, this is a refactor - the original code already exists in the master branch.

Second of all, `math.ceil` is not what `int(x + 0.5)` does. I guess you meant `round`? 

The original code was added in 114376f7344b688dca3da65ec0ad9af16935ade8 by @matejcik. Matejcik: Did you have any special reason why not to use just `round(fee_rate, 2)`? We don't use this anywhere in the codebase, so there might be a reason why not to use it, but I don't know of any.

Reply:
Ah, stupid me - the comment above explains it: "Use format_amount to get correct thousands separator  - micropython's built-in formatting doesn't add thousands sep to floating point numbers."

Sorry for the noise.

Reply:
`ceil` would round up, but we want to round to the nearest. Micropython doesn't implement `round()` AFAIK.

Reply:
It does, not in `math` but in `builtins`. Agreed that this is out of scope, apologies.

Reply:
Oh interesting, that's good to know, thanks. Anyway, this particular line of code originated [here](https://github.com/trezor/trezor-firmware/commit/114376f7344b688dca3da65ec0ad9af16935ade8), so I felt no need to change it. Maybe @matejcik can provide insight as to why `int(x+0.5)` is better than `round(x)`, but TBH I think it doesn't matter.

Reply:
i vaguely remember that `round` still returns a float. i might be wrong though